cardano-cli/src/Cardano/CLI/Commands/Legacy.hs

```diff
@@ -8,16 +8,16 @@
 module Cardano.CLI.Commands.Legacy
   ( -- * CLI command types
     LegacyCommand (..)
-  , AddressCmds (..)
-  , StakeAddressCmds (..)
-  , KeyCmds (..)
-  , TransactionCmds (..)
-  , NodeCmds (..)
-  , PoolCmds (..)
-  , QueryCmds (..)
-  , GovernanceCmds (..)
-  , GenesisCmds (..)
-  , TextViewCmds (..)
+  , LegacyAddressCmds (..)
```

Comment:
I think I dont't get this part:
>An additional change was made which is that all legacy command type names are prefix with Legacy. The intention of this is to drop the EraBased prefix from era-based command types. A prefix is necessary to avoid confusion between the two types (which is possible). It is better that it is the legacy code is prefixed rather than the era-based code because we will increasingly work with era-based code so it is better for the naming overhead to live with the legacy code.

EraBased commands aren't prefixed from what I could find.

I'm not a big fan of prefixing types with `Legacy` - module name should be enough imo. Unless we want to turn types into a constant reminder that they should be removed eventually.

Reply:
One of the reasons I opted for `Legacy` is greppability.

For example I know the type hasn't been imported indirectly via some other module without `Legacy` in its qualified name:

```
find cardano-cli/src/Cardano/CLI/EraBased -name '*.hs' | grep -v dist-newstyle | xargs grep Legacy
```

Reply:
The module name is `Legacy` so without prefixing types you should be able to do the same, because the import would be there, right?

Reply:
That is true, but imports can still pass transitively through modules that lack `Legacy` in the qualified module name.

Reply:
The module name is the big one though, so greppability of the prefix is a secondary concern in comparision, yes.

Reply:
It will be routine to import `EraBased` modules from `Legacy` modules so we could drop the `Legacy` prefix altogether at the cost of using qualified imports.

Reply:
Let's leave it in there and have a separate PR for deciding what to do with the prefix.  That way we can have a good look at the impact of different decisions with actual code.

Reply:
I prefer the Legacy prefix, it immediately tells me I'm dealing with a legacy command.